python/dgl/heterograph.py

```diff
@@ -4423,7 +4431,19 @@ def apply_edges(self, func, edges=ALL, etype=None, inplace=False):
             edata = core.invoke_gsddmm(g, func)
         else:
             edata = core.invoke_edge_udf(g, eid, etype, func)
-        self._set_e_repr(etid, eid, edata)
+
+        if self._graph.number_of_etypes() == 1 or etype is not None:
+            self._set_e_repr(etid, eid, edata)
+        else:
```

Comment:
I believe `_set_e_repr` supports more than one edata tensors stored in a dictionary. Why making the change?

Reply:
It doesn't.  The use case of feeding a dictionary in `edata` is handled by `HeteroEdgeDataView`.